From 8bb0e24c72f8fc7ef4ecd1e51e482b3fc6856945 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppymaster@gmail.com>
Date: Sun, 8 Oct 2017 16:14:45 -0400
Subject: [PATCH] test: skip hwdb and sysv-generator if the features are
 disabled (#7026)

(cherry picked from commit 1e99c1ee825fc2ba451a3f0ddf5e922bd8bd43ad)
---
 test/meson.build | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/test/meson.build b/test/meson.build
index 995a971778..c2df7ec226 100644
--- a/test/meson.build
+++ b/test/meson.build
@@ -163,9 +163,11 @@ endif
 
 ############################################################
 
-sysv_generator_test_py = find_program('sysv-generator-test.py')
-test('sysv-generator-test',
-     sysv_generator_test_py)
+if conf.get('HAVE_SYSV_COMPAT') == 1
+        sysv_generator_test_py = find_program('sysv-generator-test.py')
+        test('sysv-generator-test',
+             sysv_generator_test_py)
+endif
 
 ############################################################
 
@@ -181,6 +183,8 @@ udev_test_pl = find_program('udev-test.pl')
 test('udev-test',
      udev_test_pl)
 
-hwdb_test_sh = find_program('hwdb-test.sh')
-test('hwdb-test',
-     hwdb_test_sh)
+if conf.get('ENABLE_HWDB') == 1
+        hwdb_test_sh = find_program('hwdb-test.sh')
+        test('hwdb-test',
+             hwdb_test_sh)
+endif
